### ISSUE TITLE
True color colon syntax

### ISSUE THREAD

**Issue by mintty** [termstandard/colors#38]
About Querying the Terminal: the response should actually look like ...;48:2::1:2:3m - note the double colon which allows a position for the optional color space value.

**Comment by SlySven** [comment]
Yeah, for reference the documentation is in https://www.itu.int/rec/dologin_pub.asp?lang=e&id=T-REC-T.416-199303-I!!PDF-E&type=items take a look at page 49 of 77 (confusingly numbered **41**!) - unfortunately many interpretations forget about the second sub-parameter after the first one of 2 (or 3 or 4) and it *isn't* optional unless one is specifying a black which has all of the red, green and blue components as zero! Using a semi-colon as the sub-parameter separator is just so broken as it makes it so very hard to interpret an incoming code sequence...!

**Comment by kurahaupo** [comment]
@SlySven Generally one gets either<tt>CSI 48 ; 2 : <i>colourspace</i> : <i>red</i> : <i>green</i> : <i>blue</i> m </tt> _or_  <tt>CSI 48 ; 2 ; <i>red</i> ; <i>green</i> ; <i>blue</i> m </tt>. That is, the colourspace slot depends on which separator is used. The terminals that use `:` tend to stick more closely to the IEC spec, while those that use `;` tend to omit the colourspace.

So in practice, one can read a semicolon-separated sequence simply by looking at each element to determine how many additional elements to consume:
* If you receive 38 or 48, then take then next element as a type indicator.
  * Then if that type indicator 2 or 3 take 3 more element (for rgb or cmy);
  * if it's 4 then take 4 more elements (for cmyk);
  * if it's is 5 then take one more element (for the index).

I grant you that this isn't particularly robust, as it will fail if a new colourspace is introduced with a new code. But if I had to guess, I'd speculate that the next colourspace to be added would be RGBA, so it's probably reasonable to assume that an unknown colourspace takes 4 elements.

**Comment by mintty** [comment]
There is no hybrid syntax (neither specified nor legacy) where the first separator would be ; and the others : 
Standard syntax is colon-only, and you would need to swallow as many sub-parameters as are separated with colons, whether their number fits the type or not. Heuristics would apply to legacy semicolon syntax only.
